src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -135,7 +135,8 @@ private PerAnalyzerState GetAnalyzerState(DiagnosticAnalyzer analyzer)
             AnalyzerDriver driver,
             CancellationToken cancellationToken)
         {
-            lock (_gate)
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
❔ Why is this one checked before calling `DisposableWait`, where most of the others are not?

Reply:
Thanks, removed in https://github.com/dotnet/roslyn/pull/30593/commits/488b634ee94774d2241c6b0a12438ad0c76de87c